engine/runtime/src/test/scala/org/enso/compiler/test/pass/resolve/GenerateDocumentationTest.scala

```diff
@@ -0,0 +1,278 @@
+package org.enso.compiler.test.pass.resolve
+
+import org.enso.compiler.context.{InlineContext, ModuleContext}
+import org.enso.compiler.core.IR
+import org.enso.compiler.pass.resolve.GenerateDocumentation
+import org.enso.compiler.pass.{PassConfiguration, PassManager}
+import org.enso.compiler.test.CompilerTest
+import org.scalatest.Inside
+
+class GenerateDocumentationTest extends CompilerTest with Inside {
+
+  // === Test Setup ===========================================================
+
+  val passConfig: PassConfiguration = PassConfiguration()
+
+  implicit val passManager: PassManager =
+    new PassManager(List(), passConfig)
+
+  /** Resolves documentation comments in a module.
+    *
+    * @param ir the module
+    */
+  implicit class ResolveModule(ir: IR.Module) {
+
+    /** Resolves documentation comments for [[ir]].
+      *
+      * @param moduleContext the context in which to resolve
+      * @return [[ir]], with documentation resolved
+      */
+    def resolve(implicit moduleContext: ModuleContext): IR.Module = {
+      GenerateDocumentation.runModule(ir, moduleContext)
+    }
+  }
+
+  /** Resolves documentation comments in an expression.
+    *
+    * @param ir the expression
+    */
+  implicit class ResolveExpression(ir: IR.Expression) {
+
+    /** Resolves documentation comments for [[ir]].
+      *
+      * @param inlineContext the context in which to resolve
+      * @return [[ir]], with documentation resolved
+      */
+    def resolve(implicit inlineContext: InlineContext): IR.Expression = {
+      GenerateDocumentation.runExpression(ir, inlineContext)
+    }
+  }
+
+  /** Creates a defaulted module context.
+    *
+    * @return a defaulted module context
+    */
+  def mkModuleContext: ModuleContext = {
+    buildModuleContext()
+  }
+
+  /** Creates a defaulted inline context.
+    *
+    * @return a defaulted inline context
+    */
+  def mkInlineContext: InlineContext = {
+    buildInlineContext()
+  }
+
+  /** Gets documentation metadata from a node.
+    * Throws an exception if missing.
+    *
+    * @param ir the ir to get the doc from.
+    * @return the doc assigned to `ir`.
+    */
+  def getDoc(ir: IR): String = {
+    val meta = ir.getMetadata(GenerateDocumentation)
+    meta shouldBe defined
+    meta.get.documentation
+  }
+
+  def docGenGetAssertion(inner: String): String =
+    s"""<html>
+       | <body>
+       |   <div class="doc" style="font-size: 13px;">
+       |     <div>
+       |       <div class="">
+       |         <div class="example">
+       |           <div class="summary">$inner</div>
+       |         </div>
+       |       </div>
+       |     </div>
+       |   </div>
+       | </body>
+       |</html>""".stripMargin
+      .replaceAll(System.lineSeparator(), "")
+      .replaceAll(">[ ]+<", "><")
```

Comment:
I would consider renaming this function or adding a comment, because it is unclear to me from the name what it does.